packages/pigeon/platform_tests/test_plugin/macos/Classes/TestPlugin.swift

```diff
@@ -99,96 +101,164 @@ public class TestPlugin: NSObject, FlutterPlugin, HostIntegrationCoreApi {
     return aNullableObject
   }
 
-  func noopAsync(completion: @escaping () -> Void) {
-    completion()
+  func noopAsync(completion: @escaping (Result<Void, Error>) -> Void) {
+    completion(.success(Void()))
+  }
+
+  func echoAsync(_ aString: String, completion: @escaping (Result<String, Error>) -> Void) {
+    let result: Result<String, Error>
+    result = .success(aString)
+    completion(result)
+  }
+
+  func throwAsyncError(completion: @escaping (Result<Any?, Error>) -> Void) {
+    completion(.failure(FlutterError(code: "code", message: "message", details: "details")))
   }
 
-  func echoAsync(_ aString: String, completion: @escaping (String) -> Void) {
-    completion(aString)
+  func throwAsyncErrorFromVoid(completion: @escaping (Result<Void, Error>) -> Void) {
+    completion(.failure(FlutterError(code: "code", message: "message", details: "details")))
   }
 
-  func callFlutterNoop(completion: @escaping () -> Void) {
+  func callFlutterNoop(completion: @escaping (Result<Void, Error>) -> Void) {
     flutterAPI.noop() {
-      completion()
+      completion(.success(Void()))
     }
   }
 
-  func callFlutterEcho(_ everything: AllTypes, completion: @escaping (AllTypes) -> Void) {
-      flutterAPI.echo(everything) { completion($0) }
+  func callFlutterEcho(_ everything: AllTypes, completion: @escaping (Result<AllTypes, Error>) -> Void) {
+    flutterAPI.echo(everything) { 
+      let result: Result<AllTypes, Error>
+      result = .success($0)
+      completion(result) 
+    }
   }
 
   func callFlutterSendMultipleNullableTypes(
-    aBool aNullableBool: Bool?,
-    anInt aNullableInt: Int32?,
-    aString aNullableString: String?,
-    completion: @escaping (AllNullableTypes) -> Void
+    aBool aNullableBool: Bool?, 
+    anInt aNullableInt: Int32?, 
+    aString aNullableString: String?, 
```

Comment:
Nit: can you remove the trailing whitespace from these lines? (If your editor has an option to do it automagically I highly recommend it :) )